tests/common.py

```diff
@@ -29,3 +33,62 @@ def pytest_sim_params(metafunc, simulators=None):
                 targets.append(simulator)
 
         metafunc.parametrize('simulator', targets)
+
+def pytest_real_type_params(metafunc, real_types=None):
+    if real_types is None:
+        real_types = [RealType.FixedPoint, RealType.FloatReal, RealType.HardFloat]
+
+    if 'real_type' in metafunc.fixturenames:
+        metafunc.parametrize('real_type', real_types)
+
+class MsdslTester(fault.Tester):
+    def __init__(self, circuit, clock=None, expect_strict_default=True, debug_mode=True):
+        super().__init__(circuit=circuit, clock=clock,
+                         expect_strict_default=expect_strict_default)
+        self.debug_mode = debug_mode
+
+    def compile_and_run(self, target='system-verilog', ext_srcs=None,
+                        inc_dirs=None, ext_model_file=True, tmp_dir=None,
+                        disp_type=None, real_type=RealType.FixedPoint,
+                        defines=None, **kwargs):
+        # set defaults
+        if ext_srcs is None:
+            ext_srcs = []
+        if inc_dirs is None:
+            inc_dirs = []
+        if tmp_dir is None:
+            tmp_dir = not self.debug_mode
+        if disp_type is None:
+            disp_type = 'on_error' if (not self.debug_mode) else 'realtime'
+        if defines is None:
```

Comment:
we might want to look into doing something similar for anasymod, where a single user flag will cause **HARD_FLOAT** to be defined, and pull into the necessary header and source files.  (HardFloat should not be required, though, if the **HARD_FLOAT** option is not being used)